app/src/main/java/com/jerboa/ui/components/login/Login.kt

```diff
@@ -147,7 +150,7 @@ fun LoginForm(
                     expanded = false
                 },
             ) {
-                instanceOptions.forEach { selectionOption ->
+                instanceOptions.filter { instance.isEmpty() || it.contains(instance) }.forEach { selectionOption ->
```

Comment:
I noticed a lot of slowness and keyboard issues here, and it might be because the filtering is being done inside the ExposedDropdown.

https://alexzh.com/jetpack-compose-dropdownmenu/

Check the bottom of that page, it has an example of filtering. Very close to yours, but try it out.